src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,46 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int ARRAY_MAX_SIZE = 10;
+    private K[] keys;
+    private V[] values;
+    private int size;
+
+    public StorageImpl() {
+        this.keys = (K[]) new Object[ARRAY_MAX_SIZE];
+        this.values = (V[]) new Object[ARRAY_MAX_SIZE];
+        this.size = 0;
+    }
+
+    private int findIndexKey(K key) {
+        for (int i = 0; i < size; i++) {
+            if (keys[i] == null ? key == null : keys[i].equals(key)) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = findIndexKey(key);
+        if (index != -1) {
+            values[index] = value;
+        } else if (size < ARRAY_MAX_SIZE) {
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        }
     }
 
     @Override
     public V get(K key) {
-        return null;
+        int index = findIndexKey(key);
+        return index != -1 ? (V) values[index] : null;
```

Comment:
I'd reverse the -1 check to not make it `!=` but make it `==`, which would be easier to read, also make `-1` a constant
```suggestion
        return index == -1 ? null : (V) values[index];
```